test/integration/model/sync.test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+/* jshint -W030 */
+/* jshint -W110 */
+var chai = require('chai')
+  , Sequelize = require('../../../index')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support');
+
+describe(Support.getTestDialectTeaser('Model'), function() {
+  describe('sync', function () {
+    beforeEach(function() {
+      this.testSync = this.sequelize.define('testSync', {
+        dummy: Sequelize.STRING
+      });
+
+      return this.testSync.drop();
+    });
+
+    it('should remove a column if it exists in the databases schema but not the model', function() {
+      var User = this.sequelize.define('testSync', {
+        name: Sequelize.STRING,
+        age: Sequelize.INTEGER
+      });
+      return this.sequelize.sync()
+        .bind(this)
+        .then(function() {
+          this.sequelize.define('testSync', {
+            name: Sequelize.STRING
+          });
+        })
+        .then(() => this.sequelize.sync({alter: true}))
+        .then(function() {
+          return User.describe().then((data) => {
+            expect(data).to.not.ownProperty('age');
+          });
+        });
+    });
+
+    it('should add a column if it exists in the model but not the database', function() {
+      var testSync = this.sequelize.define('testSync', {
```

Comment:
this isn't used anywhere outside the scope and is redefined in some places. should I use let or will it work with const?